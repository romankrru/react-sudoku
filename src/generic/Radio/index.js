import React from "react";
import cn from "classnames";

import styles from "./index.module.css";

const Radio = (props) => {
	return (
		<label className={cn(styles.radio, props.className)}>
			<input
				className={styles.input}
				type="radio"
				name={props.name}
				value={props.value}
				checked={props.checked}
				onChange={props.onChange}
			/>
			<div className={styles.indicator}></div>
			{props.children}
		</label>
	);
};

export default Radio;
