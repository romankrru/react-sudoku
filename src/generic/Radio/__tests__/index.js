import React from "react";
import { render } from "@testing-library/react";

import Radio from "../";

describe("<Radio />", () => {
	it("renders without crashes", () => {
		const view = render(<Radio />);
		expect(view.container).toMatchSnapshot();
	});
});
