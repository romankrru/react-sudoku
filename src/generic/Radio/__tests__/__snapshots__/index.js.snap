// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> renders without crashes 1`] = `
<div>
  <label
    class="radio"
  >
    <input
      class="input"
      type="radio"
      value=""
    />
    <div
      class="indicator"
    />
  </label>
</div>
`;
